src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBindingProvider.cs

```diff
@@ -99,41 +101,21 @@ internal SignalRTriggerAttribute GetParameterResolvedAttribute(SignalRTriggerAtt
                 ? parameterNamesFromAttribute
                 : parameterNames;
 
-            var resolvedConnectionString = GetResolvedConnectionString(
-                typeof(SignalRTriggerAttribute).GetProperty(nameof(attribute.ConnectionStringSetting)),
-                attribute.ConnectionStringSetting);
-
-            return new SignalRTriggerAttribute(hubName, category, @event, parameterNames) {ConnectionStringSetting = resolvedConnectionString};
+            return new SignalRTriggerAttribute(hubName, category, @event, parameterNames) { ConnectionStringSetting = attribute.ConnectionStringSetting };
         }
 
-        private string GetResolvedConnectionString(PropertyInfo property, string configurationName)
+        private string GetResolvedConnectionString(string connectionStringKey)
         {
-            string resolvedConnectionString;
-            if (!string.IsNullOrWhiteSpace(configurationName))
-            {
-                resolvedConnectionString = _nameResolver.Resolve(configurationName);
-            }
-            else
-            {
-                var attribute = property.GetCustomAttribute<AppSettingAttribute>();
-                if (attribute == null)
-                {
-                    throw new InvalidOperationException($"Unable to get AppSettingAttribute on property {property.Name}");
-                }
-                resolvedConnectionString = _nameResolver.Resolve(attribute.Default);
-            }
-
-            return string.IsNullOrEmpty(resolvedConnectionString)
-                ? _options.ConnectionString
-                : resolvedConnectionString;
+            var monitor = (_managerStore.GetOrAddByConnectionStringKey(connectionStringKey) as ServiceHubContextStore).OptionsMonitor;
```

Comment:
I prefer add a `ConnectionString` property to `IServiceHubContextStore`. It will make this more general, especially in test. As the `_managerStore` is a interface, we can mock it and generate a mocked `IServiceHubContextStore` in test.

Reply:
We want dynamic configuration so we need `IOptionsMonitor` to get the runtime value of `ServiceManagerOptions`. How about add `IOptionsMonitor` as an internal get-only property to `IServiceHubContextStore`?

Reply:
What about adding `public string ConnectionString {get;}` to interface and implement `public string ConnectionString => optionsMonitor.CurrentValue.ConnectionString`. Thus you can still get the latest value. Seems exposing the whole IOptionsMonitor is not necessary.